apps/examples/testcase/ta_tc/systemio/itc/itc_gpio.c

```diff
@@ -302,26 +302,21 @@ static void itc_systemio_gpio_register_unregister_callback_p(void)
 	ret = iotbus_gpio_set_direction(g_gpio_h2, IOTBUS_GPIO_DIRECTION_IN);
 	TC_ASSERT_EQ_CLEANUP("iotbus_gpio_set_direction", ret, 0, iotbus_gpio_close(g_gpio_h2); iotbus_gpio_close(g_gpio_h));
 
-	ret = iotbus_gpio_register_cb(g_gpio_h2, IOTBUS_GPIO_EDGE_BOTH, gpio_event_callback, (void *)g_gpio_h2);
+	ret = iotbus_gpio_register_cb(g_gpio_h2, IOTBUS_GPIO_EDGE_RISING, gpio_event_callback, (void *)g_gpio_h2);
 	TC_ASSERT_EQ_CLEANUP("iotbus_gpio_register_cb", ret, 0, iotbus_gpio_close(g_gpio_h2); iotbus_gpio_close(g_gpio_h));
 
 	// To trigger event for callback
 	ret = iotbus_gpio_write(g_gpio_h, data);
 	TC_ASSERT_EQ_CLEANUP("iotbus_gpio_write", ret, 0, iotbus_gpio_close(g_gpio_h2); iotbus_gpio_close(g_gpio_h));
 	sleep(WAIT_SECONDS);
 
-	TC_ASSERT_EQ_CLEANUP("iotbus_gpio_register_cb not triggered", gpio_flag_callback, 1, iotbus_gpio_unregister_cb(g_gpio_h2); iotbus_gpio_close(g_gpio_h2); iotbus_gpio_close(g_gpio_h));
-	TC_ASSERT_EQ_CLEANUP("iotbus_gpio_register_cb response", gpio_callback_data, data, iotbus_gpio_unregister_cb(g_gpio_h2); iotbus_gpio_close(g_gpio_h2); iotbus_gpio_close(g_gpio_h));
-
-	gpio_flag_callback = 0;
```

Comment:
1. Because there are several tests before below, initialization of gpio_flag_callback is right to confirm changes.
2. You don't check any result of write operation (line 309). Then why do you call it? You set gpio irq as rising edge and set gpio as 0. Let's check un-called status of gpio_event_callback, because 0 doesn't make edges of gpio.

Reply:
@sunghan-chang 
Fixed as you suggested. Please review

Reply:
Could you check my first comment? I don't know why you remove gpio_flag_callback = 0;.

Reply:
@sunghan-chang 
gpio_flag_callback  is already initialized to 0 at starting of testcase. 
later at line 313, gpio_flag_callback is compared to 0, if it is not 0, then ASSERT will FAIL.
Then at line 316, "iotbus_gpio_write" api is called which invokes gpio_callback, and now gpio_flag_callback  is compared to 1.
Please suggest if still "gpio_flag_callback = 0" this is required, I will insert it.